tests/workflow_task.go

```diff
@@ -960,3 +961,117 @@ func (s *FunctionalSuite) TestWorkflowTerminationSignalAfterTransientWorkflowTas
   6 WorkflowTaskScheduled
   7 WorkflowExecutionTerminated`, s.getHistory(s.namespace, we))
 }
+
+func (s *FunctionalSuite) TestAddTask_BacklogInfo() {
+	id := uuid.New()
+	wt := "functional-workflow-workflow-task-heartbeating-local-activities"
+	tl := id
+	identity := "worker1"
+
+	workflowType := &commonpb.WorkflowType{Name: wt}
+
+	tq := &taskqueuepb.TaskQueue{Name: tl, Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	request := &workflowservice.StartWorkflowExecutionRequest{
+		RequestId:           uuid.New(),
+		Namespace:           s.namespace,
+		WorkflowId:          id,
+		WorkflowType:        workflowType,
+		TaskQueue:           tq,
+		Input:               nil,
+		WorkflowRunTimeout:  durationpb.New(20 * time.Second),
+		WorkflowTaskTimeout: durationpb.New(3 * time.Second),
+		Identity:            identity,
+	}
+
+	resp0, err0 := s.engine.StartWorkflowExecution(NewContext(), request)
+	s.NoError(err0)
+
+	we := &commonpb.WorkflowExecution{
+		WorkflowId: id,
+		RunId:      resp0.RunId,
+	}
+
+	s.EqualHistoryEvents(`
+  1 WorkflowExecutionStarted
+  2 WorkflowTaskScheduled`, s.getHistory(s.namespace, we))
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	s.Eventually(func() bool {
+		resp, err := s.engine.DescribeTaskQueue(ctx, &workflowservice.DescribeTaskQueueRequest{
+			Namespace:              s.namespace,
+			TaskQueue:              &taskqueuepb.TaskQueue{Name: tl, Kind: enumspb.TASK_QUEUE_KIND_NORMAL},
+			ApiMode:                enumspb.DESCRIBE_TASK_QUEUE_MODE_ENHANCED,
+			Versions:               nil, // default version, in this case unversioned queue
+			TaskQueueTypes:         nil, // both types
+			ReportPollers:          true,
+			ReportTaskReachability: true,
+			ReportBacklogInfo:      true,
+		})
+		s.NoError(err)
+		s.NotNil(resp)
+		s.Assert().Equal(1, len(resp.GetVersionsInfo()), "should be 1 because only default/unversioned queue")
+		versionInfo := resp.GetVersionsInfo()[""]
+		s.Assert().Equal(enumspb.BUILD_ID_TASK_REACHABILITY_REACHABLE, versionInfo.GetTaskReachability())
+
+		addedTasks := int64(1)
+		checker := true
+		for queueType, t := range versionInfo.GetTypesInfo() {
+			if queueType == 1 {
+				if t.BacklogInfo.ApproximateBacklogCount != addedTasks {
```

Comment:
could you add comments why you use `!=` and `>` in the bellow checks? not clear why

Reply:
The reason why I was using `!=` is because tasks are only being added to the workflow task queue and not the activity based one.  I shall now be removing the `>` and replacing that with a check of having the backlog for the activity queue to be 0. (something such as -> )

`
else if queueType == 2 {
				if t.BacklogInfo.ApproximateBacklogCount != 0 {
					checker = false
				}
`